Text/TextAnalyticsForHealth/TextAnalyticsForHealth.cs

```diff
@@ -0,0 +1,189 @@
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using AzureCognitiveSearch.PowerSkills.Common;
+using Azure.AI.TextAnalytics;
+using Azure;
+using System.Globalization;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.TextAnalyticsForHealth
+{
+    public static class TextAnalyticsForHealth
+    {
+        public static readonly string textAnalyticsApiEndpointSetting = "TEXT_ANALYTICS_API_ENDPOINT";
+        public static readonly string defaultTextAnalyticsEndpoint = "https://centralus.api.cognitive.microsoft.com";
+        public static readonly string textAnalyticsApiKeySetting = "TEXT_ANALYTICS_API_KEY";
+        private static readonly int defaultTimeout = 230;
+        private static readonly int maxTimeout = 230;
+        private static readonly int timeoutBuffer = 5;
+        private static readonly int maxCharLength = 5000;
+
+        [FunctionName("TextAnalyticsForHealth")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log,
+            ExecutionContext executionContext)
+        {
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            // Get Endpoint and access key from App Settings
+            string apiKey = Environment.GetEnvironmentVariable(textAnalyticsApiKeySetting, EnvironmentVariableTarget.Process);
+            string apiEndpoint = Environment.GetEnvironmentVariable(textAnalyticsApiEndpointSetting, EnvironmentVariableTarget.Process);
+            if (apiEndpoint == null)
+            {
+                apiEndpoint = defaultTextAnalyticsEndpoint;
+            }
+            if (apiKey == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Healthcare Text Analytics API key is missing. Make sure to set it in the Environment Variables.");
+            }
+            var client = new TextAnalyticsClient(new Uri(apiEndpoint), new AzureKeyCredential(apiKey));
+
+            // Get a custom timeout from the header, if it exists. If not use the default timeout.
+            int timeout;
+            if (!int.TryParse(req.Headers["SkillTimeout"].ToString(), out timeout))
+            {
+                timeout = defaultTimeout;
+            }
+            timeout = Math.Clamp(timeout - timeoutBuffer, 0, maxTimeout);
+            var timeoutMiliseconds = timeout * 1000;
+            var timeoutTask = Task.Delay(timeoutMiliseconds);
+
+            // Get a custom default language, if none is provided, use english
+            string defaultLanguage = req.Headers.ContainsKey("DefaultLanguage")? req.Headers["DefaultLanguage"].ToString(): "en";
```

Comment:
>DefaultLanguage [](http://example.com/codeflow?start=62&length=15)

Let's call this "defaultLanguageCode" (with the camel instead of pascal casing) to match what we call it for built in skills